From 5d8ae14ab0a17f1f89d47231a7ad7a362ddf21f8 Mon Sep 17 00:00:00 2001
From: Francesco Rollo <eferollo@gmail.com>
Date: Thu, 28 Aug 2025 10:56:12 +0000
Subject: [PATCH] feat(5.26 diverr_2): Integer division errors

Signed-off-by: Francesco Rollo <eferollo@gmail.com>
---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 33 +++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..0354e58 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -616,6 +616,39 @@ static __always_inline int syncookie_handle_syn(struct header_pointers *hdr,
 	if (hdr->tcp->fin || hdr->tcp->rst)
 		return XDP_DROP;
 
+	/*
+	 * Get a tainted integer from the packet data.
+	 * `hdr->tcp->ack_seq` is a __u16.
+	 * It can legitimately have a value of 0 in a TCP header.
+	 * This makes it a perfect candidate for a tainted divisor
+	 * that can cause divide-by-zero.
+	 */
+	__u16 tainted_divisor_val = hdr->tcp->ack_seq;
+
+	/* Define a non-zero numerator for our operations. */
+	int numerator_val = 100;
+
+	/*
+	 * Perform a division operation where the divisor
+	 * (`tainted_divisor_val`) can be zero. The eBPF verifier is
+	 * expected to diagnose this. -> From docs:
+	 *
+	 * If eBPF program execution would result in division by zero,
+	 * the destination register is instead set to zero. If execution
+	 * would result in modulo by zero, for BPF_ALU64 the value of
+	 * the destination register is unchanged whereas for BPF_ALU
+	 * the upper 32 bits of the destination register are zeroed.
+	 */
+	int result_division = numerator_val / tainted_divisor_val;
+
+	/*
+	 * Use bpf_printk to reference the results. This ensures the
+	 * operations are not optimized away by the compiler, and forces
+	 * the verifier to fully analyze them.
+	 */
+	bpf_printk("[diverr]: Divisor: %u, Div Result: %d",
+		   hdr->tcp->ack_seq, result_division);
+	
 	/* Issue SYN cookies on allowed ports, drop SYN packets on blocked
 	 * ports.
 	 */
-- 
2.43.0

