From b00cf2fc705c1806c3a43844e9d8e2c75dafca1e Mon Sep 17 00:00:00 2001
From: Giorgio <s331003@studenti.polito.it>
Date: Sat, 19 Jul 2025 15:17:41 +0000
Subject: [PATCH] feat(5_30 intoflow) Overflowing signed integers

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..dd8e5f8 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -6,6 +6,7 @@
 #include <bpf/bpf_helpers.h>
 #include <bpf/bpf_endian.h>
 #include <asm/errno.h>
+#include <sys/cdefs.h>
 
 #define TC_ACT_OK 0
 #define TC_ACT_SHOT 2
@@ -753,6 +754,19 @@ static __always_inline int syncookie_handle_ack(struct header_pointers *hdr)
 	return XDP_PASS;
 }
 
+//Test start
+
+/* Objtive test if verifier can check overflow for tainted input value
+ * We use as tainted value the sequence number 
+ */
+
+static __always_inline void checkOverflow(int value)
+{
+	bpf_printk("Initial value %d", value);
+	int result = value + 2147483647; //This would cause overflow if value > 0
+	bpf_printk("Overflow here %d", result);
+}
+
 static __always_inline int syncookie_part1(void *ctx, void *data, void *data_end,
 					   struct header_pointers *hdr, bool xdp)
 {
@@ -766,6 +780,9 @@ static __always_inline int syncookie_part1(void *ctx, void *data, void *data_end
 	if (ret != XDP_TX)
 		return ret;
 
+	//test site
+	checkOverflow(bpf_htons(hdr->tcp->seq)); //seq number tend to be high so guaranteed overflow
+
 	/* Packet is TCP and doesn't belong to an established connection. */
 
 	if ((hdr->tcp->syn ^ hdr->tcp->ack) != 1)
-- 
2.43.0

