From 3c02181e02962b627026518a63bcde4e533ba059 Mon Sep 17 00:00:00 2001
From: s331003 <s331003@studenti.polito.it>
Date: Tue, 26 Aug 2025 14:29:00 +0000
Subject: [PATCH] feat(5.35b uninitmem)

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..b52b9dc 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -753,11 +753,35 @@ static __always_inline int syncookie_handle_ack(struct header_pointers *hdr)
 	return XDP_PASS;
 }
 
+//Start of test
+
+static __always_inline void uninitialized_packet_read(void *ctx, void *data_end)
+{
+    const int add_len = 64;
+    
+    // Dynamically increase packet size, exposing uninitialized memory
+    if (bpf_xdp_adjust_tail(ctx, add_len) != 0) {
+        return; // Failed to grow packet, nothing to do
+    }
+
+    void *new_tail_ptr = data_end;
+    
+    // Access the newly exposed, uninitialized memory
+    for(int i = 0; i < add_len; i++){
+        char uninit_byte = ((char *)new_tail_ptr)[i];
+        bpf_printk("Uninitialized byte %d: %x\n", i, uninit_byte);
+    }
+    return;
+}
+
 static __always_inline int syncookie_part1(void *ctx, void *data, void *data_end,
 					   struct header_pointers *hdr, bool xdp)
 {
 	int ret;
 
+	//test launch
+	uninitialized_packet_read(ctx, data_end);
+
 	ret = tcp_dissect(data, data_end, hdr);
 	if (ret != XDP_TX)
 		return ret;
-- 
2.43.0

