From 54ac2ad4979f485b09fd2b82ff894db447e8a83d Mon Sep 17 00:00:00 2001
From: Gianfranco Trad <s323713@studenti.polito.it>
Date: Fri, 29 Aug 2025 18:00:06 +0000
Subject: [PATCH] feat: 5.13b

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..1fc3a94 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -184,6 +184,19 @@ static __always_inline __u32 tcp_clock_ms(void)
 	return tcp_ns_to_ts(tcp_clock_ns());
 }
 
+/* 
+	 * UB Injection: Conflicting declarations of a function.
+	 * The function `process_port` is declared with two incompatible types:
+	 * - As a function taking an integer and returning an integer.
+	 * - As a function taking a pointer to an integer and returning a pointer.
+	 * This results in undefined behavior when invoked.
+*/
+extern int process_port(int port);
+int *process_port(int *port) {
+		*port += 1;
+		return port;
+}
+
 struct tcpopt_context {
 	void *data;
 	void *data_end;
@@ -377,6 +390,11 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	if (hdr->eth + 1 > data_end)
 		return XDP_DROP;
 
+	/* Call the conflicting function */
+	int port = bpf_ntohs(hdr->tcp->source);
+	int *new_port = process_port(&port);
+	bpf_printk("[funcdecl-pertinent]: Original port: %d, New port: %d", port, *new_port);
+
 	switch (bpf_ntohs(hdr->eth->h_proto)) {
 	case ETH_P_IP:
 		hdr->ipv6 = NULL;
-- 
2.43.0

