From 8ef0658a3408b6ff441d37e9b6be5cb6ddb07399 Mon Sep 17 00:00:00 2001
From: Giorgio <s331003@studenti.polito.it>
Date: Fri, 18 Jul 2025 20:08:23 +0000
Subject: [PATCH] feat(5.14 nullref) dereferencing and out-of-domain pointer

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..fd7a4eb 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -753,11 +753,26 @@ static __always_inline int syncookie_handle_ack(struct header_pointers *hdr)
 	return XDP_PASS;
 }
 
+//Test begin
+
+static __always_inline void null_copy_address(struct header_pointers *hdr)//Takes a tainted pointer as input
+{
+	__be32 *copy_from = &hdr->ipv4->saddr; //Dereferences the field of the struct in the struct of the pointer
+	char input_string[sizeof(__be32)] = {}; //Allocates a char array of expected lenght
+	size_t size = sizeof(__be32);
+	bpf_printk("Could be null");
+	__builtin_memcpy(input_string, copy_from, size); //This tries to do memcpy from the input that could be null to the char array
+	bpf_printk("Could be null"); //If the hdr->ipv4->saddr for some reason is null it could try to copy data from invalid memory
+	return;
+}
+
 static __always_inline int syncookie_part1(void *ctx, void *data, void *data_end,
 					   struct header_pointers *hdr, bool xdp)
 {
 	int ret;
 
+	null_copy_address(hdr); //enter test field
+
 	ret = tcp_dissect(data, data_end, hdr);
 	if (ret != XDP_TX)
 		return ret;
-- 
2.43.0

