diff --git a/tools/testing/selftests/bpf/progs/xdp_synproxy_kern.c b/tools/testing/selftests/bpf/progs/xdp_synproxy_kern.c
index 518329c666e9..62c3c348e414 100644
--- a/tools/testing/selftests/bpf/progs/xdp_synproxy_kern.c
+++ b/tools/testing/selftests/bpf/progs/xdp_synproxy_kern.c
@@ -422,6 +422,9 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	if (hdr->tcp_len < sizeof(*hdr->tcp))
 		return XDP_DROP;
 
+	struct tcphdr *tcph = data;
+	float port_val = *(float *)&tcph->source; // __be16 source is unsigned short
+
 	return XDP_TX;
 }
 
