From 6817a0b5f7c8512197bb3f1717cbe005ccf7205b Mon Sep 17 00:00:00 2001
From: Gianfranco Trad <s323713@studenti.polito.it>
Date: Wed, 23 Jul 2025 13:17:38 +0000
Subject: [PATCH] feat: (5.22 invptr) using oob pointers or array subscripts

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..60d4f52 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -421,7 +421,12 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	hdr->tcp_len = hdr->tcp->doff * 4;
 	if (hdr->tcp_len < sizeof(*hdr->tcp))
 		return XDP_DROP;
+	/* === UB Injection (5.22: invalid pointer value) === */
+	int ub_offset = -MAX_ALLOWED_PORTS;
+	__u16 *value = bpf_map_lookup_elem(&allowed_ports, &ub_offset);  // UB
+	__u16 ub_trigger = *value;  // optional dereference
 
+	(void)ub_trigger;
 	return XDP_TX;
 }
 
-- 
2.43.0

