From 4df2e2f7e9ef1f9deef0883b269238981109c430 Mon Sep 17 00:00:00 2001
From: Gianfranco Trad <s323713@studenti.polito.it>
Date: Fri, 29 Aug 2025 19:22:35 +0000
Subject: [PATCH] feat: 5.24a

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..346e475 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -377,6 +377,19 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	if (hdr->eth + 1 > data_end)
 		return XDP_DROP;
 
+/* 
+	 * UB Injection: Tainted format string from a map.
+	 * A format string is retrieved from a BPF map, which could contain
+	 * user-controlled or unvalidated data. Using this string in a formatted
+	 * output function like `bpf_printk` introduces undefined behavior.
+*/
+	__u32 key = hdr->ipv4->daddr; // Example key based on destination IP
+	char *tainted_fmt = bpf_map_lookup_elem(&values, &key); // Retrieve tainted format string
+	if (tainted_fmt) {
+		/* Using the tainted format string directly in bpf_printk */
+		bpf_printk(tainted_fmt);
+	}
+
 	switch (bpf_ntohs(hdr->eth->h_proto)) {
 	case ETH_P_IP:
 		hdr->ipv6 = NULL;
-- 
2.43.0

