From 5332a0297f57b4c2e715ea4156cf0dc784407100 Mon Sep 17 00:00:00 2001
From: s331003 <s331003@studenti.polito.it>
Date: Tue, 26 Aug 2025 08:10:43 +0000
Subject: [PATCH] feat(5.15a) addrescape

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..ef7705b 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -753,11 +753,25 @@ static __always_inline int syncookie_handle_ack(struct header_pointers *hdr)
 	return XDP_PASS;
 }
 
+// ===== Test: Example 2 =====
+// Returning pointer to local automatic object (array)
+static __always_inline int *init_array(void)
+{
+	int array[5] = { 1, 2, 3, 4, 5 };
+	return array; // diagnostic required
+}
+
 static __always_inline int syncookie_part1(void *ctx, void *data, void *data_end,
 					   struct header_pointers *hdr, bool xdp)
 {
 	int ret;
 
+	// Trigger Example 2
+	int *arr_ptr = init_array();
+	if (arr_ptr) {
+		bpf_printk("Leaked array[0]: %d", arr_ptr[0]);
+	}
+
 	ret = tcp_dissect(data, data_end, hdr);
 	if (ret != XDP_TX)
 		return ret;
-- 
2.43.0

