From f4f16cd84a6a726267f580f10b1f4b6dc3ce307e Mon Sep 17 00:00:00 2001
From: Gianfranco Trad <s323713@studenti.polito.it>
Date: Fri, 29 Aug 2025 18:55:59 +0000
Subject: [PATCH] feat: 5.22e

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..52774e7 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -377,6 +377,11 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	if (hdr->eth + 1 > data_end)
 		return XDP_DROP;
 
+	/* UB Injection: Pointer to just past the end of an object */
+	__u8 *tcp_header_end = (__u8 *)hdr->tcp + hdr->tcp_len; 
+	__u8 *invalid_access = tcp_header_end + 1;             
+	bpf_printk("[invptr-4]: Invalid access pointer: %p", invalid_access);
+
 	switch (bpf_ntohs(hdr->eth->h_proto)) {
 	case ETH_P_IP:
 		hdr->ipv6 = NULL;
-- 
2.43.0

