From 53cdd18a1ef51d5d2823c52334e0c2a8bf5521cd Mon Sep 17 00:00:00 2001
From: Francesco Rollo <eferollo@gmail.com>
Date: Thu, 28 Aug 2025 20:49:43 +0000
Subject: [PATCH] feat(5.26 diverr_5): Integer division errors

Signed-off-by: Francesco Rollo <eferollo@gmail.com>
---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..a2d68cb 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -616,6 +616,19 @@ static __always_inline int syncookie_handle_syn(struct header_pointers *hdr,
 	if (hdr->tcp->fin || hdr->tcp->rst)
 		return XDP_DROP;
 
+	/*
+	 * Modulo result not representable
+	 * Case: INT_MIN % -1 -> overflow UB in two’s complement
+	 */
+
+	/* Consider the following value as tainted */
+	int tainted_dividend = 0x80000000;
+	int divisor_neg_one = -1;
+
+	int result_overflow = tainted_dividend / divisor_neg_one;
+	bpf_printk("[diverr][overflow]: dividend=%d divisor=%d result=%d",
+		   tainted_dividend, divisor_neg_one, result_overflow);
+
 	/* Issue SYN cookies on allowed ports, drop SYN packets on blocked
 	 * ports.
 	 */
-- 
2.43.0

