From 6815fef4e8cb7dd0c21bf32b829524cd3c72d366 Mon Sep 17 00:00:00 2001
From: Francesco Rollo <eferollo@gmail.com>
Date: Thu, 28 Aug 2025 20:43:56 +0000
Subject: [PATCH] feat(5.26 diverr_4): Integer division errors

Signed-off-by: Francesco Rollo <eferollo@gmail.com>
---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..add3707 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -616,6 +616,19 @@ static __always_inline int syncookie_handle_syn(struct header_pointers *hdr,
 	if (hdr->tcp->fin || hdr->tcp->rst)
 		return XDP_DROP;
 
+	/*
+	 * Division result not representable
+	 * Case: INT_MIN / -1 -> overflow UB in two’s complement
+	 */
+
+	/* Consider this value as tainted */	
+	int tainted_dividend = 0x80000000;
+	int divisor_neg_one = -1;
+
+	int result_overflow = tainted_dividend / divisor_neg_one;
+	bpf_printk("[diverr][overflow]: dividend=%d divisor=%d result=%d",
+		   tainted_dividend, divisor_neg_one, result_overflow);
+
 	/* Issue SYN cookies on allowed ports, drop SYN packets on blocked
 	 * ports.
 	 */
-- 
2.43.0

