From 7fd035e2a0680339f30a728385d2028e4a2e9b16 Mon Sep 17 00:00:00 2001
From: Gianfranco Trad <s323713@studenti.polito.it>
Date: Fri, 29 Aug 2025 17:26:29 +0000
Subject: [PATCH] feat: 5.01c

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..c2522d9 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -376,6 +376,17 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	hdr->eth = data;
 	if (hdr->eth + 1 > data_end)
 		return XDP_DROP;
+/* 
+	 * UB Injection 3: Accessing hdr->eth as a uint32_t pointer.
+	 * This violates strict aliasing rules because hdr->eth is a pointer
+	 * to a struct ethhdr, but we access it as a uint32_t pointer.
+	 * This results in undefined behavior.
+*/
+	__u32 *fake_eth = (__u32 *)hdr->eth; 
+	__u32 eth_value = *fake_eth;       
+
+	bpf_printk("[ptrcomp]: Accessed eth_value: %u", eth_value);
+
 
 	switch (bpf_ntohs(hdr->eth->h_proto)) {
 	case ETH_P_IP:
-- 
2.43.0

