From 469dc330b2465e2c5e691edf24acf8ae4c16f066 Mon Sep 17 00:00:00 2001
From: Gianfranco Trad <s323713@studenti.polito.it>
Date: Fri, 29 Aug 2025 18:53:41 +0000
Subject: [PATCH] feat: 5.22d

---
 XDPs/xdp_synproxy/xdp_synproxy_kern.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/XDPs/xdp_synproxy/xdp_synproxy_kern.c b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
index 0d0d990..1b36feb 100644
--- a/XDPs/xdp_synproxy/xdp_synproxy_kern.c
+++ b/XDPs/xdp_synproxy/xdp_synproxy_kern.c
@@ -377,6 +377,11 @@ static __always_inline int tcp_dissect(void *data, void *data_end,
 	if (hdr->eth + 1 > data_end)
 		return XDP_DROP;
 
+	/* UB Injection 4: Array subscript out of range */
+	__u16 tcp_array_info[3] = {hdr->tcp->source, hdr->tcp->dest, hdr->tcp_len};
+	__u16 out_of_bounds_value = tcp_array[5];
+	bpf_printk("[invptr-3]: Out-of-bounds array value: %u", out_of_bounds_value);
+
 	switch (bpf_ntohs(hdr->eth->h_proto)) {
 	case ETH_P_IP:
 		hdr->ipv6 = NULL;
-- 
2.43.0

